types/json-api-serializer/index.d.ts

```diff
@@ -58,23 +58,48 @@ declare namespace JSONAPISerializer {
 
   interface LinkObject {
     href: string;
-    meta: unknown;
+    meta?: Meta;
   }
 
   interface LinksObject {
-    [name: string]: LinkObject | LinksCallback | string | null;
+    self?: LinkObject | LinksCallback | string | null;
+    related?: LinkObject | LinksCallback | string | null;
   }
 
   interface ResourceObject<T> {
     id: string;
     type: string;
     attributes?: Omit<T, 'id'> | undefined;
     relationships?: {
-      [key: string]: { data: ResourceObject<any> | Array<ResourceObject<any>> };
+      [key: string]: Relationship;
     } | undefined;
     links?: LinksObject | LinksCallback | undefined;
   }
 
+  type Relationship = {
+    links?: LinksObject | LinksCallback | undefined;
+    data: Linkage | Linkage[];
+    meta?: Meta;
+  } | {
+      links?: LinksObject | LinksCallback | undefined;
+      data?: Linkage | Linkage[];
+      meta: Meta;
+  } | {
+      links: LinksObject | LinksCallback | undefined;
+      data?: Linkage | Linkage[];
+      meta?: Meta;
+  };
+
+  interface Meta {
+    [name: string]: any;
```

Comment:
I prefer to use `unknown` due to its stricter nature. It's harder to encounter runtime errors since you are forced to check for properties. Is it ok to use unknown instead?

Alternatively you could use a union type of the primitive types, since that is the only types that can be transferred over JSON.

Reply:
sure